contracts/eosio.system/src/eosio.system.cpp

```diff
@@ -500,6 +555,67 @@ namespace eosiosystem {
       return false;
    }
 
+   // -------------------------------------------------------------------------------------------
+   struct canon_name_t
+   {
+      uint64_t _value; // all significant characters are shifted to the least significant position
+      uint64_t _size;  // number of significant characters
+
+      uint64_t mask() const { return (1ull << (_size * 5)) - 1; }
+      uint64_t size() const { return _size; }
+
+      // starts from the end, so n[0] is the last significant character of the name
+      uint64_t operator[](uint64_t i) const { return (_value >> (5 * i)) & 0x1F; }
+
+      canon_name_t(name n) {
+         uint64_t v         = n.value;
+         auto     num_zeros = std::countr_zero(v);
+         check(num_zeros >= 4, "Account names and patterns should be less than 13 characters long");
```

Comment:
This check occurs when constructing the `canon_name_t` which currently only occurs in a call to `name_allowed` which is only called when checking the new account name against patterns in `newaccount`.

But what if bad patterns are already in the table at this point?
Then this check will fail preventing any new accounts from being created (unless created by eosio which bypasses the name restriction checks).

We need to prevent the bad patterns from ever entering the table in the first place. That means in `add_patterns_to` you need to make sure the pattern is not 13 characters long. More generally, you need to ensure it satisfied all the requirements assumed by `canon_name_t`.

Reply:
https://github.com/eosnetworkfoundation/eos-system-contracts/pull/189/commits/b9a3391c02b817921bafa1f385a007dd911041b8

Reply:
https://github.com/eosnetworkfoundation/eos-system-contracts/pull/189/commits/8aaef04c6c09baca2490c02dd9b08ae0c75df0aa